src/App.ts

```diff
@@ -1091,7 +1092,11 @@ export default class App<AppCustomContext extends StringIndexed = StringIndexed>
 
     // Get the client arg
     let { client } = this;
-    const token = selectToken(context);
+
+    // If functionBotAccessToken exists on context, the incoming event is function-related *and* the
+    // user has `attachFunctionToken` enabled. In that case, subsequent calls with the client should
+    // use the function-related/JIT token in lieu of the botToken or userToken.
+    const token = context.functionBotAccessToken ? context.functionBotAccessToken : selectToken(context);
```

Comment:
I considered moving this to the section below ([here](https://github.com/slackapi/bolt-js/pull/2200/files#diff-6532e508d489a84018e407c57238c87c5b74221e2de2589e4ce8f0c281f0f5f7L1098)), but left it for clarity and to only have a single site in `processEvent` where the `token` is set.

I also didn't nest it within `selectToken` since `selectToken` is used by another utility method (`say`), which seemed like a good thing to keep entirely separate.

I'm not married to where it sits, so happy to relocate it if anyone feels strongly about alternatives.

Reply:
Thanks for adding the detailed comment. This change sounds reasonable to me too